department-of-veterans-affairs/abd-vro
# MCP-2352: Apply "authoredOn" helper function to v1 templates only

## What was the problem?

Helper function was trying to parse "authoredOn" date, which can be empty for OCR data. It is only needed for the version 1 templates.

Associated tickets or Slack threads:

- [MCP-2352](https://amida.atlassian.net/browse/MCP-2352)

## How does this fix it?[^1]

Add template version to the template variables json. Only parse "authoredOn"  for version 1 templates. 

## How to test this PR
- Step 1 build and run app
- Step 2 For generate-pdf endpoint, set "pdfTemplate" to "v2" and set a medication "authoredOn" to empty string ("")
- Step 3 send request and make sure there is no error
- [^secrel]


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
